### PR TITLE
[Documentation] Update parameter and function attribute section in LangRef

### PR DESCRIPTION
**Pull request by rajatbajpai**
Update the documentation for parameter and function attributes to include support for target-dependent string attributes.

### CODE REVIEW

**Comment by rajatbajpai** [comment]
Can someone please take a look at this change? Thanks!

File: llvm/docs/LangRef.rst [hunk 1]
```diff
@@ -1161,15 +1161,17 @@ parameters of a function. Parameter attributes are considered to be part
 of the function, not of the function type, so functions with different
 parameter attributes can have the same function type.
 
-Parameter attributes are simple keywords that follow the type specified.
-If multiple parameter attributes are needed, they are space separated.
-For example:
+Parameter attributes are simple keywords that follow the specified type.
+When multiple parameter attributes are required, they are separated by
+spaces. Additionally, target-dependent attributes can be provided as a
```
Comment (nikic) [inline llvm/docs/LangRef.rst:1166]: I don't think the terminology "target-dependent attributes" is really in use anymore, these are simply "string attributes". We have plenty of target-independent string attributes as well.

Comment (rajatbajpai) [reply llvm/docs/LangRef.rst:1166]: Thanks, Nikita. I have updated the changes.

**Review by nikic — APPROVED** [review]
LGTM

**Comment by rajatbajpai** [comment]
Thank you, Nikita!

**Pull request merged**